### ISSUE TITLE
Setting VideoFileClip's Duration to longer than the original file causes OSError

### ISSUE THREAD

**Issue by HC-47** [Zulko/moviepy#1522]
I wanted to increase the duration of a video (reducing the fps accordingly), but if I try this:

```

from moviepy.video.io.VideoFileClip import VideoFileClip
pause = VideoFileClip(os.path.normpath('eara.mp4')) #duration == 2s
pause_2 = pause.set_duration(3)
pause_3 = pause_2.set_fps(pause_2.fps/3*2) 
```

I get this error:

```
  File "<stdin>", line 1, in <module>
  File "D:\dropbox\Dropbox (Personale)\ASTEROIDI ANALI\video musicati\codice_pulito\sing_sing_sing.py", line 208, in create_new_video
    video.write_videofile(self.video_name+'.mp4', write_logfile=True)
  File "<decorator-gen-55>", line 2, in write_videofile
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "<decorator-gen-54>", line 2, in write_videofile
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 135, in use_clip_fps_by_default
    return f(clip, *new_a, **new_kw)
  File "<decorator-gen-53>", line 2, in write_videofile
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 22, in convert_masks_to_RGB
    return f(clip, *a, **k)
  File "C:\Python36\lib\site-packages\moviepy\video\VideoClip.py", line 298, in write_videofile
    logger=logger)
  File "<decorator-gen-45>", line 2, in write_audiofile
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "C:\Python36\lib\site-packages\moviepy\audio\AudioClip.py", line 210, in write_audiofile
    logger=logger)
  File "<decorator-gen-9>", line 2, in ffmpeg_audiowrite
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "C:\Python36\lib\site-packages\moviepy\audio\io\ffmpeg_audiowriter.py", line 169, in ffmpeg_audiowrite
    logger=logger):
  File "C:\Python36\lib\site-packages\moviepy\audio\AudioClip.py", line 86, in iter_chunks
    fps=fps, buffersize=chunksize)
  File "<decorator-gen-44>", line 2, in to_soundarray
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 54, in requires_duration
    return f(clip, *a, **k)
  File "C:\Python36\lib\site-packages\moviepy\audio\AudioClip.py", line 127, in to_soundarray
    snd_array = self.get_frame(tt)
  File "<decorator-gen-11>", line 2, in get_frame
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "C:\Python36\lib\site-packages\moviepy\Clip.py", line 93, in get_frame
    return self.make_frame(t)
  File "C:\Python36\lib\site-packages\moviepy\audio\AudioClip.py", line 297, in make_frame
    for c, part in zip(self.clips, played_parts)
  File "C:\Python36\lib\site-packages\moviepy\audio\AudioClip.py", line 298, in <listcomp>
    if (part is not False)]
  File "<decorator-gen-11>", line 2, in get_frame
  File "C:\Python36\lib\site-packages\moviepy\decorators.py", line 89, in wrapper
    return f(*new_a, **new_kw)
  File "C:\Python36\lib\site-packages\moviepy\Clip.py", line 93, in get_frame
    return self.make_frame(t)
  File "C:\Python36\lib\site-packages\moviepy\audio\io\AudioFileClip.py", line 77, in <lambda>
    self.make_frame = lambda t: self.reader.get_frame(t)
  File "C:\Python36\lib\site-packages\moviepy\audio\io\readers.py", line 172, in get_frame
    "with clip duration=%d seconds, "%self.duration)
OSError: Error in file eara.mp4, Accessing time t=2.04-2.09 seconds, with clip duration=2 seconds,
```

Now, I do understand the error (the original clip does not have a 3 s duration), but I think that the increase in FPS should conceptually do the trick. Am I missing something?

**Comment by tburrows13** [comment]
Reducing the fps like that won't play the video slower, it will just make it less smooth. `set_fps` has a `change_duration` parameter which you can set to True to do what you want. I think it's pretty new though, so probably only available in the 2.0 pre-releases.

**Comment by HC-47** [comment]
Thanks for your help!
